### ISSUE TITLE
Conversion from *Input to *FluentBuilder

### ISSUE THREAD

**Issue by robjtede** [awslabs/aws-sdk-rust#1004]
### Describe the feature

An easy way to convert from `*Input`s to `*FluentBuilder`s or a way to use an `*Input` as a request as-is.

(I suspect the former is easier to accomplish and less of a semver hazard.)

### Use Case

When an `*Input` is constructed elsewhere and passed into a function that starts constructing the `*FluentBuilder` using the S3 client, we are forced to deconstruct the already valid Input into the `.set_*()` methods provided which is error prone.

### Proposed Solution

Using Rust RFC format:

```
impl ListObjectsV2FluentBuilder {
  pub fn from_input(input: ListObjectsV2Input) -> Self;
}
```

and etc. for other operations.

I'm mostly familiar with the S3 crate but imagine this applies to a wider scope.

### Acknowledgements

- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### Community Note
* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue, please leave a comment

**Comment by rcoh** [comment]
Very reasonable! If you want to take a stab at it yourself, the relevant code is [here](https://github.com/smithy-lang/smithy-rs/blob/572c5d7650fba40b232516e51ca8dde5952a22df/codegen-client/src/main/kotlin/software/amazon/smithy/rust/codegen/client/smithy/generators/client/FluentClientGenerator.kt#L342-L348)

For your use case, would you want a `From` impl, or would something like `.set_input()` or `input_mut()` work?

**Comment by robjtede** [comment]
Exactly. Our use case uses the Input and Output types in a trait interface that abstracts over in-memory, disk, and S3 "backends" but always trying to respect S3 parameter semantics.

Any of those alternative solutions would work for us, too.

**Comment by jdisanti** [comment]
I don't know if this helps your situation, but all input builder types have a `send_with` method: https://docs.rs/aws-sdk-s3/latest/aws_sdk_s3/operation/list_objects_v2/builders/struct.ListObjectsV2InputBuilder.html#method.send_with

**Comment by robjtede** [comment]
Almost. Would be enough if there was either:
- a way to convert an Input into an InputBuilder
- a similar `Input::send_with()`

**Comment by robjtede** [comment]
Luckily I have control of the interface so I decided to change everything to InputBuilders which does solve my issue.
